kong/pdk/service/request.lua

```diff
@@ -417,6 +419,49 @@ local function new(self)
   end
 
 
+  if ngx.config.subsystem == "http" then
+    local set_upstream_cert_and_key =
+        require("resty.kong.tls").set_upstream_cert_and_key
+
+    ---
+    -- Sets the client certificate used while handshaking with the Service.
+    --
+    -- The `chain` argument is the client certificate and intermediate chain (if any)
+    -- returned by functions such as [ngx.ssl.parse\_pem\_cert](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#parse_pem_cert).
+    --
+    -- The `key` argument is the private key corresponding to the client certificate
+    -- returned by functions such as [ngx.ssl.parse\_pem\_priv\_key](https://github.com/openresty/lua-resty-core/blob/master/lib/ngx/ssl.md#parse_pem_priv_key).
+    --
+    -- @function kong.service.request.set_cert_key
```

Comment:
As for the function name itself, I think `set_client_cert_key` is a good compromise? (the "ssl" name is a historical artifact after all, we even use `"tls"` in the `protocol` fields of our entities).

Reply:
`kong.service` sounds good to me :+1:

I feel somewhat strongly that the methods dealing with downstream/upstream SSL should have `ssl` somewhere in their name though (e.g. `cosocket:sslhandshake()`), but not a blocker either.

Reply:
I don't mind it, but the `ssl` name is a historical relic at this point (newer stuff seems to avoid it, e.g. "mutual TLS"; the TLS library in Rust is called `rustls`, not `rustssl`; etc.). `cosocket:sslhandshake()` was a name created long ago. I only suggested that because I grepped the PDK and we're not using `ssl` anywhere. I would think `cert` in the name is sufficient as an identifier, but I don't feel strongly about this.

Reply:
I agree with @hishamhm, since this is the PDK function which the intention is to wrap around underlying OpenResty API, `kong.service. set_tls_client_cert_key` sounds good to me.

Reply:
Yes, `tls` instead of `ssl` is fine.